File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter04/Ternary.java
package xyz.mattyb.openwest.kotlinprimer.chapter04;

import java.util.Random;

public class Ternary {
    public static void main(String[] args) {
        int x = rand();
        String outcome = x > 75 ? "Awesome Sauce" : "Not too shabby";

        System.out.println(x + " is " +outcome);

        String maybe = maybeNull();

        String result = maybe == null ? "Better luck next time" : maybe;

        System.out.println(result);
    }

    private static int rand() {
        return new Random().nextInt(100);
    }

    private static String maybeNull() {
        int rand = rand();
        if (rand % 2 == 0) {
            return null;
        }
        return "Totes not null";
    }
}

File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter04/ternary.kt
package xyz.mattyb.openwest.kotlinprimer.chapter04

import java.util.*

fun main(args: Array<String>) {
    /**
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     * Don't want to give away how functions work ;-)
     */
    val x = rand()
    val outcome = if (x > 75) "Awesome Sauce" else "Not too shabby"

    println("$x is $outcome")

    val maybe = maybeNull()

    val result = maybe ?: "Better luck next time"

    println(result)
}

/**
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 *
 * Don't want to give away how functions work ;-)
 */
private fun rand(): Int {
    return Random().nextInt(100)
}

private fun maybeNull() : String? {
    if (rand() % 2 == 0) {
        return null;
    }
    return "Totes not null"
}

